src/configuration.rs

```diff
@@ -476,14 +389,6 @@ impl TryFrom<PluginConfiguration> for FilterConfig {
                         }
                     }
                 }
-                for action in &rule.actions {
-                    for datum in &action.data {
-                        let result = datum.item.compile();
-                        if result.is_err() {
-                            return Err(result.err().unwrap());
-                        }
-                    }
-                }
```

Comment:
Actually one thing that's standing out to me is that we no longer keep the path as well as compiled version for PatternExpressions and don't pre-compile data items. I'm not 100%, does this conflict somewhat with the intention [here](https://github.com/Kuadrant/wasm-shim/issues/112#issuecomment-2419152322) cc @alexsnaps

Reply:
I need to look into those changes properly, but that would also mean that we'd err out at evaluation time instead of at config parse time... that seems... not desirable 🤔

Reply:
I removed because I did not see any point in early compilation. The "compilation" is just about building the path out of the selector string, which can never fail. Hence, I removed. 

Can you confirm my analysis? 

Anyway, I am happy to add that back.

Reply:
Doing it once, instead of on all invocations? Now we need to parse on every [`Policy::pattern_expression_applies`](https://github.com/Kuadrant/wasm-shim/pull/111/files#diff-016defdf85b98ea93c8767b2cbdce1d6eb7d8a1d0fe3c62a68e22e201c65c0efR61), right? And it will possibly fail when we move these to CEL expression as @adam-cattermole pointed out. But arguably that's my problem. 

In any case, I'm a big proponent of doing things as early as possible, represent them in the proper way from there on in the system and do the least amount of work. This code was early parsing the "path" and keeping it in proper form to avoid paying the computation time on the data plane's path. These are my guiding principle in general, but more so here.

Reply:
Thanks for confirming.

Bringing that back